Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/PartitionSceneEntitiesSystem.cs

```diff
@@ -26,144 +23,125 @@ namespace ECS.SceneLifeCycle.Systems
-            if (definition.IsEmpty)
+            PartitionScenes partitionScenes = new ()
             {
-                PartitionComponent partitionComponent = partitionComponentPool.Get();
-                // some default values to not break other systems
-                partitionComponent.Bucket = emptyScenePartition;
-                World.Add(entity, partitionComponent);
-                return;
-            }
-
-            if (definition.IsPortableExperience)
-            {
-                PartitionComponent partitionComponent = partitionComponentPool.Get();
-                partitionComponent.OutOfRange = false;
-                partitionComponent.Bucket = 0;
-                partitionComponent.IsBehind = false;
-                partitionComponent.RawSqrDistance = 1;
-                partitionComponent.IsDirty = true;
-                World.Add(entity, partitionComponent);
-                return;
-            }
-
-            if (definition.InternalJobIndex < 0)
-            {
-                ScheduleSceneDefinition(ref definition);
-            }
-            else
-            {
-                PartitionComponent partitionComponent = partitionComponentPool.Get();
-                partitionDataContainer.SetPartitionComponentData(definition.InternalJobIndex, ref partitionComponent);
-                World.Add(entity, partitionComponent);
-            }
-        }
-
-        protected void ScheduleSceneDefinition(ref SceneDefinitionComponent definition)
-        {
-            AddCorners(ref definition);
-
-            var partitionData = new PartitionData
-            {
-                IsDirty = readOnlyCameraSamplingData.IsDirty, RawSqrDistance = -1
+                CameraPosition = readOnlyCameraSamplingData.Position,
+                CameraForward = readOnlyCameraSamplingData.Forward,
+                CameraIsDirty = readOnlyCameraSamplingData.IsDirty,
+                PartitionPool = partitionComponentPool,
+                UnloadingSqrDistance = unloadingDistance * unloadingDistance,
+
+                // PartitionSettingsAsset.SqrDistanceBuckets is a List, so just cast to that so we don't
+                // waste time on virtual calls when accessing its elements.
+                DistanceBuckets = (List<int>)partitionSettings.SqrDistanceBuckets,
             };
-            partitionDataContainer.SetPartitionData(partitionData);
-        }
 
-        protected void AddCorners(ref SceneDefinitionComponent definition)
-        {
-            var corners = new NativeArray<ParcelCorners>(definition.ParcelsCorners.Count, Allocator.Persistent);
+            Profiler.BeginSample(nameof(PartitionScenes));
 
-            for (var i = 0; i < definition.ParcelsCorners.Count; i++)
-                corners[i] = definition.ParcelsCorners[i];
+            World.InlineQuery<PartitionScenes, SceneDefinitionComponent, PartitionComponent>(
+                in existingScenesQuery, ref partitionScenes);
```

Comment:
So this takes 0.12 ms to run on my laptop. If I change it to InlineParallelQuery, it takes 1 ms to run. Crazy.